hypothesis-python/docs/reference/developer.rst

```diff
@@ -0,0 +1,29 @@
+Developer Reference
+===================
```

Comment:
I'm concerned that this will be ambiguous - all our users are also developers, in the general sense.  How about something like "Hypothesis internals" or "Unstable & Internals"?

Reply:
`Hypothesis internals` overlaps with the future "contributor / internals guide", but I think it's the best option